waypoint/services/nats_service.py

```diff
@@ -25,44 +37,63 @@ def __init__(self, jetstream: JetStreamContext):
         self.js_context: JetStreamContext = jetstream
 
     async def _subscribe(
-        self, *, group_id: str, wallet_id: str, topic: str, state: str, look_back: int
+        self,
+        *,
+        group_id: str,
+        wallet_id: str,
+        topic: str,
+        state: str,
+        start_time: str = None,
     ) -> JetStreamContext.PullSubscription:
-        try:
-            logger.trace(
-                "Subscribing to JetStream for wallet_id: {}, group_id: {}",
-                wallet_id,
-                group_id,
-            )
-            group_id = group_id or "*"
-            subscribe_kwargs = {
-                "subject": f"{NATS_STATE_SUBJECT}.{group_id}.{wallet_id}.{topic}.{state}",
-                "stream": NATS_STATE_STREAM,
-            }
 
-            # Get the current time in UTC
-            current_time = datetime.now(timezone.utc)
+        logger.debug(
+            "Subscribing to JetStream for wallet_id: {}, group_id: {}",
+            wallet_id,
+            group_id,
+        )
 
-            # Subtract look_back time from the current time
-            look_back_time = current_time - timedelta(seconds=look_back)
+        group_id = group_id or "*"
+        subscribe_kwargs = {
+            "subject": f"{NATS_STATE_SUBJECT}.{group_id}.{wallet_id}.{topic}.{state}",
+            "stream": NATS_STATE_STREAM,
+        }
 
-            # Format the time in the required format
-            start_time = look_back_time.strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
-            config = ConsumerConfig(
-                deliver_policy=DeliverPolicy.BY_START_TIME,
-                opt_start_time=start_time,
-            )
-            subscription = await self.js_context.pull_subscribe(
-                config=config, **subscribe_kwargs
-            )
+        config = ConsumerConfig(
+            deliver_policy=DeliverPolicy.BY_START_TIME,
+            opt_start_time=start_time,
+        )
 
-            return subscription
+        @tenacity.retry(
+            retry=tenacity.retry_if_exception_type(TimeoutError),
+            wait=tenacity.wait_exponential(multiplier=1, max=16),
+            after=retry_log,
+            stop=tenacity.stop_never,
```

Comment:
just for neatness, we can import all these `tenacity.*` members from the package, so we can reference them without the `tenacity.` 

Also perhaps add a comment to summarise: "this decorator will retry indefinitely if TimeoutError occurs, with exponential back-off between retries". Something like that